### DISCUSSION TITLE
Window Length within make_reduction

### DISCUSSION THREAD

**Discussion by edzWH** [alan-turing-institute/sktime#6554] [Q&A]
I'm currently using XGBoost as a regressor, utilising the make_reduction function.

I have a question regarding the window_length param. I understand this the length of the sliding window, with x past values being used to forecast y.

Say I have a rare feature which only appears once a year within a set of daily data, but has high importance. This feature will appear multiple times in the full training set as I have multiple years data. If the sliding window length isn't long enough to see one of these rare features at the point of forecast will the model not have seen the feature and have trained on it?

For example if I want the model to have seen the annual feature, will I need to set the window length to at least 365 if I'm using daily data.

Thanks.

**Comment by fkiraly** [comment]
@edzWH, I don't think so, the condition for the model to see/learn the rare features would rather be, it needs to end up in a couple of windows covering multiple occurrences.

This can very well happen if the window is shorter than a year.
